examples/platform/silabs/wifi/icd/ApplicationSleepManager.h

```diff
@@ -0,0 +1,138 @@
+        return *this;
+    }
+
+    ApplicationSleepManager & SetCommissioningWindowManager(chip::CommissioningWindowManager * commissioningWindowManager)
+    {
+        mCommissioningWindowManager = commissioningWindowManager;
+        return *this;
+    }
+
+    ApplicationSleepManager & SetWifiSleepManager(chip::DeviceLayer::Silabs::WifiSleepManager * wifiSleepManager)
+    {
+        mWifiSleepManager = wifiSleepManager;
+        return *this;
+    }
+
+    // ReadHandler::ApplicationCallback implementation
+
+    /**
+     * @brief Calls the WifiSleepManager VerifyAndTransitionToLowPowerMode.
+     *        The VerifyAndTransitionToLowPowerMode function is responsible of then queriyng the ApplicationSleepManager to
+     *        determine in which low power state the Wi-Fi device can transition to.
+     */
+    void OnSubscriptionTerminated(chip::app::ReadHandler & aReadHandler);
+
+    /**
+     * @brief Calls the WifiSleepManager VerifyAndTransitionToLowPowerMode.
+     *        The VerifyAndTransitionToLowPowerMode function is responsible of then queriyng the ApplicationSleepManager to
+     *        determine in which low power state the Wi-Fi device can transition to.
+     */
+    void OnSubscriptionEstablished(chip::app::ReadHandler & aReadHandler);
+
+    CHIP_ERROR OnSubscriptionRequested(chip::app::ReadHandler & aReadHandler, chip::Transport::SecureSession & aSecureSession);
+
+    // WifiSleepManager::ApplicationCallback implementation
+
+    /**
+     * @brief TODO
+     *
+     * @return true
+     * @return false
+     */
+    bool CanGoToLIBasedSleep() override;
+
+    // FabricTable::Delegate implementation
+
+    /**
+     * @brief Calls the WifiSleepManager VerifyAndTransitionToLowPowerMode.
+     *        The VerifyAndTransitionToLowPowerMode function is responsible of queriyng the ApplicationSleepManager to
+     *        determine in which low power state the Wi-Fi device can transition to.
+     */
+    void OnFabricRemoved(const chip::FabricTable & fabricTable, chip::FabricIndex fabricIndex) override;
+
+    /**
+     * @brief Calls the WifiSleepManager VerifyAndTransitionToLowPowerMode.
+     *        The VerifyAndTransitionToLowPowerMode function is responsible of queriyng the ApplicationSleepManager to
+     *        determine in which low power state the Wi-Fi device can transition to.
+     */
+    void OnFabricCommitted(const chip::FabricTable & fabricTable, chip::FabricIndex fabricIndex) override;
+
+    void OnFabricUpdated(const chip::FabricTable & fabricTable, chip::FabricIndex fabricIndex) override {}
+
+private:
+    ApplicationSleepManager()  = default;
+    ~ApplicationSleepManager() = default;
+
+    ApplicationSleepManager(const ApplicationSleepManager &)             = delete;
+    ApplicationSleepManager & operator=(const ApplicationSleepManager &) = delete;
+
+    static ApplicationSleepManager mInstance;
+    chip::FabricTable * mFabricTable                                  = nullptr;
```

Comment:
All the Fabric callback here have a reference to the fabric table, do we really need to keep a pointer to it? can't we use that reference instead?